### PR TITLE
Dnn-8448: Support host-level OAuth authentication provider configuration

### PR DESCRIPTION
**Pull request by JoshuaBradley**
Pull request for https://dnntracker.atlassian.net/browse/DNN-8448

### CODE REVIEW

File: DNN Platform/Library/Services/Authentication/AuthenticationController.cs [hunk 1]
```diff
@@ -268,6 +274,32 @@ public static List<AuthenticationInfo> GetEnabledAuthenticationServices()
...
+                                     || a.AuthenticationType.Equals("Live")
+                                     || a.AuthenticationType.Equals("Twitter"))
+                                  ? IsEnabledForPortal(a, PortalSettings.Current.PortalId)
+                                  : !string.IsNullOrEmpty(a.LoginControlSrc) && ((control?.LoadControl("~/" + a.LoginControlSrc) as AuthenticationLoginBase)?.Enabled ?? true)
+                    where !a.AuthenticationType.Equals("DNN") && enabled
+                    select a).Any();
+        }
+
+        /// <summary>
+        /// Determines whether [is enabled for portal] [the specified authentication].
```
Comment (bdukes) [inline DNN Platform/Library/Services/Authentication/AuthenticationController.cs]: @JoshuaBradley can you fix the doc comments (and add some to the above method)?  Thanks!

File: DNN Platform/Modules/CoreMessaging/Subscriptions.ascx.cs [hunk 2]
```diff
@@ -40,6 +40,8 @@
 
 namespace DotNetNuke.Modules.CoreMessaging
 {
+    using DotNetNuke.Services.Authentication;
```
Comment (bdukes) [inline DNN Platform/Modules/CoreMessaging/Subscriptions.ascx.cs]: Can you move this with the other `using` statements?

**Comment by bdukes** [comment]
:+1:  looks good to me, once those couple of comments are addressed.  thanks!

**Comment by JoshuaBradley** [comment]
The comments have been addressed. Thanks!

**Comment by bdukes** [comment]
@jbrinkman @ashishpd is there interest in this PR?

**Comment by wmorgenweck** [comment]
@bdukes Definitely interest on this one, but I think it will require updates for DNN9, correct?

**Comment by bdukes** [comment]
that's right @wmorgenweck.

@JoshuaBradley, can you check the box on this pull request to let dnnsoftware contributors edit your PR?  Thanks!

**Comment by galatrash-at-dnn** [comment]
@JoshuaBradley this PR has conflicts. To be able to review, you need to go and recreate it again against the latest codebase. I will close this now without merging and feel free to submit another PR after resolving the conflicts.

**Comment by bdukes** [comment]
@galatrash-at-dnn I've rebased and removed conflicts

**Comment by bdukes** [comment]
@ohine I've fixed these conflicts

**Pull request merged**